packages/engine/Source/Scene/PropertyTable.js

```diff
@@ -296,24 +294,21 @@ PropertyTable.prototype.getPropertyIds = function (index, results) {
 
   if (defined(this._metadataTable)) {
     // concat in place to avoid unnecessary array allocation
-    results.push.apply(
-      results,
-      this._metadataTable.getPropertyIds(scratchResults)
-    );
+    for (let i = 0; i < this._metadataTable.propertyIds.length; i++) {
```

Comment:
Thanks for the feedback. I am currently going through it. Would you have time for a 1:1 chat about the codebase?

Reply:
Depending on what exactly will be discussed, it may make sense for someone else to join that call/chat as well, but we can sort that out, probably via mail `marco`@`cesium`.`com`.

Reply:
> Depending on what exactly will be discussed, it may make sense for someone else to join that call/chat as well, but we can sort that out, probably via mail `marco`@`cesium`.`com`.

Ok thank you. Sorry about the delay. We just graduated our program, oh and we won first place on best capstone project!